### PR TITLE
Animaltrain

### PR DESCRIPTION
**Pull request by mrrobinhood5**
### Summary
New DT Animal Training

#### PR Type

- [ ] This PR fixes an existing alias
- [x] This PR creates a new alias
#### Other
- [x] Changes have been tested

### CODE REVIEW

File: Aliases/Server - Heroism & Shame/dt/animal list.alias [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+          f' -f "Fix| Run `!dt animal setup` to save your animals." '
+else:
+  savedAnimals = load_json(DTAnimalTrain)
+returnstr += f' -title "{me.name} checks their saved Animals"'
+for animal, data in savedAnimals.items():
+    ccName = "Animal Training-"+animal
+    returnstr += f' -f "{animal}| Name: `{data["name"]}` \n' \
+                 f' Current Level: `{data["traininglevel"]}` \n '
+    if data["traininglevel"] == "drilled":
+        returnstr += f' Train Progress: `COMPLETE!`" '
```
Comment (1drturtle) [inline Aliases/Server - Heroism & Shame/dt/animal list.alias:15]: Looks like you are adding a field with no closing `"`

File: Aliases/Server - Heroism & Shame/dt/animal setup.alias [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    cr = "1"
+
+
+#checks if you can train the animal
+canTrain = True if int(cr) <= floor(me.levels.total_level/2) else False
+if canTrain == False:
+    return f' -title "{me.name} attempts Animal Train Setup" -f "Oops|You can\'t train that animal at your level. Darnit." '
+#checks and sets your current animal training level
+if hasPrevTraining:
+    currentLevel = pargs.get("train")[0]
```
Comment (1drturtle) [inline Aliases/Server - Heroism & Shame/dt/animal setup.alias]: `pargs.last('train')`

File: Aliases/Server - Heroism & Shame/dt/animal setup.alias [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+           f' -f "-animal|Type in the type of animal you are about to train. This can be \\"monkey\\" or \\"constrictor boa\\"." '\
+           f' -f "-cr|No quotes, this is the number of the CR. 1/2 or 2 or whatever" '\
+           f' -f "-name|*Optional* - This can be the name of the Animal if you have a name. Use quotes for names with spaces" '\
+           f' -f "-train|*Optional* - Either tamed or trained, if the animal has already had previous training or you bought it tamed." '\
+           f' -f "WARNING|Running this command after starting training will reset your training counters for that animal" '
+
+# Error Checks missing things before running the setup
+if not choseAnimal or not choseCR:
+    return setupstr
+if hasPrevTraining and pargs.get("train")[0] not in ["tamed", "trained"]:
```
Comment (1drturtle) [inline Aliases/Server - Heroism & Shame/dt/animal setup.alias]: pargs.last('train')

File: Aliases/Server - Heroism & Shame/dt/animal setup.alias [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    return setupstr
+animal, animalName, cr = pargs.get("animal")[0].lower(),pargs.get("name")[0] if setName else "nameless", pargs.get("cr")[0],
```
Comment (1drturtle) [inline Aliases/Server - Heroism & Shame/dt/animal setup.alias]: pargs.last('xxx')

File: Aliases/Server - Heroism & Shame/dt/animal.alias [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+# sets primary variables
+me, pargs, returnstr, trainComplete, trainCounter = character(), argparse(&ARGS&), "", False, 0
+
+# sets the booleans
+withAdv = True if pargs.get("adv") else False
+hasAnimal = True if exists("DTAnimalTrain") else False
+choseAnimal = True if pargs.get("name") else False
+
+# sets roll string
+rollStr = "1d20+" if not withAdv else "2d20kh1+"
```
Comment (1drturtle) [inline Aliases/Server - Heroism & Shame/dt/animal.alias]: `rollStr = ['1d20+', '2d20kh1+', '2d20kl1+'][pargs.adv()]`

This automates `adv/dis` parsing, as `pargs.adv()` returns 0 for normal, 1 for adv and 2 or -1 for dis (-1 being the last element in an array)

File: Aliases/Server - Heroism & Shame/dt/animal.alias [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        returnstr += f' -f "COMPLETE| Training Complete, will not roll anymore."'
+        trainComplete = True
+        break
+    #builds the roll and checks to see if it passed
+    animalroll = vroll(rollStr)
+    crit = animalroll.result.crit
+    critstr = "Fail"
+    passedRoll = animalroll.total >= dc
+    if passedRoll and not trainComplete:
+        if not crit == -1:
```
Comment (1drturtle) [inline Aliases/Server - Heroism & Shame/dt/animal.alias:62]: There is no crit_type with -1

File: Aliases/Server - Heroism & Shame/dt/community.alias [hunk 7]
```diff
@@ -1,4 +1,4 @@
-tembed -title "Downtime: Community Service" -footer "!help dt community | by MrRobinhood5#1999" {{pargs=&ARGS&}}
+tembed -title "Downtime: Community Service" -thumb <image> -footer "!help dt community | by MrRobinhood5#1999" {{pargs=&ARGS&}}
```
Comment (1drturtle) [inline Aliases/Server - Heroism & Shame/dt/community.alias]: `-color <color>`

**Review by 1drturtle — CHANGES_REQUESTED** [review]
Requested some changes for best practices/bugs

File: Aliases/Server - Heroism & Shame/dt/animal list.alias (hunk 1 above)
Comment (1drturtle) [reply Aliases/Server - Heroism & Shame/dt/animal list.alias:15]: Nevermind, missed the `else`

File: Aliases/Server - Heroism & Shame/dt/animal.alias (hunk 6 above)
Comment (mrrobinhood5) [reply Aliases/Server - Heroism & Shame/dt/animal.alias:62]: bruh i copied your example from embodiment. it does work it trues on 0

**Review by 1drturtle — APPROVED** [review]
- [x] Yeet

**Pull request merged**